dataland-keycloak/start_keycloak.sh

```diff
@@ -13,10 +13,17 @@ if [[ "$mode" == initialize ]]; then
   rm $(grep -E -l '"username" : "data_(reader|uploader|admin)"' "$dataland_realm_folder"/datalandsecurity-users-*.json) || echo "No technical users to be cleaned up"
   rm $(grep -E -l '"username" : "test_user.*@dataland.com"' "$dataland_realm_folder"/datalandsecurity-users-*.json) || echo "No test users to be cleaned up"
   cp /keycloak_realms/datalandsecurity-realm.json $dataland_realm_folder
-  for variable in $(env | grep KEYCLOAK_ | cut -d'=' -f1); do
-    # If one of the env variables containing "KEYCLOAK_" contains a % character the below sed statement will fail
-    # and the script will terminate because % is the sed delimiter
-    sed s%\$\{"$variable"\}%"${!variable}"%g -i $dataland_realm_folder/datalandsecurity-realm.json
+  realm_file=$dataland_realm_folder/datalandsecurity-realm.json
+  for variable in $(grep -oP '\$\{([A-Z_]+)\}' $realm_file | cut -d'{' -f2 | cut -d'}' -f1); do
+    if env | grep -eq "^$variable="; then
```

Comment:
das ist nicht ganz optimal weil man sich damit auf echte env-vars beschränkt - normale variablen sind dann raus. 

Ich habe in verifyEnvironmentVariables.sh (siehe LKSG-Branch) folgendes gemacht:
```
  value=${!var:+isSet}
  if [[ $value != "isSet" ]]; then echo Warning - $var is not set; fi
```

was meint Ihr?

Reply:
value könnte dann in dem zweiten if-block auch benutzt werden.

Reply:
Dieses Skript läuft ja innerhalb eines Docker-Containers. Daher würde ich erwarten das Variablen in aller Regel über env kommen. Das wäre nur ein Problem wenn man sie innerhalb dieses Skriptes setzen würde. Was ich als eher ungeschickt (und auch unwahrscheinlich) sehen würde.

Reply:
Ich hätte jetzt gedacht, dass man hier ausdrücklich sogar will, dass die Variablennamen aus dem realm.json gerade in den Umgebungsvariablen liegen sollen, da das ja die angestrebte Strategie ist, um das realm-template zu befüllen.